http-clients/src/test/java/org/triplea/http/client/web/socket/WebSocketConnectionTest.java

```diff
@@ -54,60 +67,142 @@ void onError() {
       webSocketConnection.getClient().onError(exception);
       verify(webSocketConnectionListener).handleError(exception);
     }
+
+    @Test
+    @DisplayName("Queued messages are flushed on connection open")
+    void queuedMessagesAreFlushedOnConnectionOpen() {
+      // not connected, this message should be queued
+      webSocketConnection.sendMessage(MESSAGE);
+
+      // onOpen will trigger message send, expect the send attempt to trigger an exception
+      // because we never connected.
+      assertThrows(
+          WebsocketNotConnectedException.class, () -> webSocketConnection.getClient().onOpen(null));
+    }
   }
 
   @ExtendWith(MockitoExtension.class)
   @Nested
   class SendMessageAndConnect {
     @Mock private WebSocketClient webSocketClient;
+    @Mock private Consumer<String> errorHandler;
 
     private WebSocketConnection webSocketConnection;
 
     @BeforeEach
     void setup() {
       webSocketConnection = new WebSocketConnection(INVALID_URI);
       webSocketConnection.setClient(webSocketClient);
+    }
+
+    @AfterEach
+    void tearDown() {
+      webSocketConnection.getPingSender().cancel();
+    }
+
+    @Test
+    @DisplayName("Verify connect initiates connection and starts the pinger")
+    void connectWillInitiateConnection() throws Exception {
+      givenWebSocketConnects(true);
+
+      webSocketConnection.connect(errorHandler);
+
+      verifyConnectWasCalled();
+      verifyPingerIsStarted();
+    }
+
+    private void verifyConnectWasCalled() throws Exception {
+      verify(webSocketClient, timeout(2000))
+          .connectBlocking(
+              WebSocketConnection.DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    }
+
+    private void verifyPingerIsStarted() {
+      assertThat(
+          "Pinger should be started on successful connection",
+          webSocketConnection.getPingSender().isRunning(),
+          is(true));
+    }
 
-      // set connect timeout to a minimum value supported for Awaitility
-      webSocketConnection.setConnectTimeoutMillis(101);
+    private void givenWebSocketConnects(final boolean connects) throws Exception {
+      when(webSocketClient.connectBlocking(
+              WebSocketConnection.DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS))
+          .thenReturn(connects);
     }
 
     @Test
-    void connect() {
-      webSocketConnection.connect();
+    @DisplayName("Verify connect failing invokes error handler and pinger is not running")
+    void connectionFailure() throws Exception {
+      givenWebSocketConnects(false);
+
+      webSocketConnection.connect(errorHandler);
+
+      // minimal amount of sleep to allow new thread to spawn up and connection attempt to fail.
+      Thread.sleep(50L);
```

Comment:
You could make the connect method return a `Future`, which would allow you to await it in this test, which avoids this ugly sleep.
Same for the other method a couple of lines below

Reply:
A unit test that is 100ms faster (best case) is very valuable, with thousands of tests, 100ms is actually a really slow test. Though, we don't quite have a 100% unit test...

Anyways, this is a good suggestion and also allows us to verify the connect return result as well for a stronger test. Updated: 3100d056d